storage/remote/queue_manager.go

```diff
@@ -851,23 +970,22 @@ func (s *shards) sendSamples(ctx context.Context, samples []prompb.TimeSeries, b
 
 // sendSamples to the remote storage with backoff for recoverable errors.
 func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.TimeSeries, buf *[]byte) error {
-	req, highest, err := buildWriteRequest(samples, *buf)
+	// Build the WriteRequest with no metadata.
+	req, highest, err := buildWriteRequest(samples, nil, *buf)
 	if err != nil {
 		// Failing to build the write request is non-recoverable, since it will
 		// only error if marshaling the proto to bytes fails.
 		return err
 	}
 
-	backoff := s.qm.cfg.MinBackoff
 	reqSize := len(*buf)
 	sampleCount := len(samples)
 	*buf = req
-	try := 0
 
 	// An anonymous function allows us to defer the completion of our per-try spans
 	// without causing a memory leak, and it has the nice effect of not propagating any
 	// parameters for sendSamplesWithBackoff/3.
-	attemptStore := func() error {
+	attemptStore := func(try int) error {
```

Comment:
did this need to be passed in?

Reply:
It does, it tells the trace which attempt are we currently on, which is incremented every time we backoff.